File: typecast-java/src/main/java/com/neosapience/exceptions/InternalServerException.java
package com.neosapience.exceptions;

/**
 * Exception thrown when the API returns a 500 Internal Server Error.
 * 
 * <p>This indicates a server-side error.</p>
 */
public class InternalServerException extends TypecastException {
    
    /**
     * Creates a new InternalServerException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public InternalServerException(String message, String responseBody) {
        super(message, 500, responseBody);
    }

    /**
     * Creates a new InternalServerException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public InternalServerException(String message, String responseBody, Throwable cause) {
        super(message, 500, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/InternalServerException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class InternalServerException extends TypecastException
{
    public function __construct(string $message = 'Internal server error', ?\Throwable $previous = null)
    {
        parent::__construct($message, 500, $previous);
    }
}
